### ISSUE TITLE
[announce] GitHub Discussions does not honor dry-run setting

### ISSUE THREAD

**Issue by charphi** [jreleaser/jreleaser#1065]
### Steps to Reproduce

Add a GitHub Discussion announce in the config without adding the `write:discussion` scope to the GitHub token.

### Expected Behaviour

JReleaser should not announce the release and raise an exception or notify about the missing scope.

### Actual Behaviour

JReleaser doesn't fail nor send notification.

### Environment Information

- **Operating System**:  Ubuntu
- **JReleaser Version:**  1.3.1
- **JDK Version:** JDK17

**Comment by aalmiray** [comment]
Are there any warnings or errors in `trace.log`?

**Comment by charphi** [comment]
Here is the trace:

```
[INFO]    [discussions] title: Sandbox Bom 2.2.8-SNAPSHOT released!
[DEBUG]   [discussions] message: 🚀 Sandbox Bom 2.2.8-SNAPSHOT has been released! https://github.com/nbbrd/sandbox/releases/tag/early-access
org.kohsuke.github.GHFileNotFoundException: https://api.github.com/organizations/8778468/team/7043227/discussions {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/teams#create-a-discussion"}
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:480)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:407)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:355)
	at org.kohsuke.github.Requester.fetch(Requester.java:76)
	at org.kohsuke.github.AbstractBuilder.done(AbstractBuilder.java:108)
	at org.kohsuke.github.GHDiscussionBuilder.done(GHDiscussionBuilder.java:78)
	at org.kohsuke.github.GHDiscussion$Creator.done(GHDiscussion.java:191)
	at org.jreleaser.sdk.github.Github.createDiscussion(Github.java:246)
	at org.jreleaser.sdk.github.DiscussionsAnnouncer.announce(DiscussionsAnnouncer.java:93)
	at org.jreleaser.engine.announce.Announcers.announce(Announcers.java:103)
	at org.jreleaser.engine.announce.Announcers.announce(Announcers.java:90)
	at org.jreleaser.workflow.AnnounceWorkflowItem.doInvoke(AnnounceWorkflowItem.java:39)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at org.jreleaser.engine.hooks.CommandHookExecutor.execute(CommandHookExecutor.java:52)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:102)
	at org.jreleaser.maven.plugin.JReleaserFullReleaseMojo.execute(JReleaserFullReleaseMojo.java:148)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: java.io.FileNotFoundException: https://api.github.com/organizations/8778468/team/7043227/discussions
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:2048)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:2043)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:2042)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1609)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1589)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:224)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient$HttpURLConnectionResponseInfo.bodyStream(GitHubHttpUrlConnectionClient.java:174)
	at org.kohsuke.github.GitHubResponse$ResponseInfo.getBodyAsString(GitHubResponse.java:314)
	at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:92)
	at org.kohsuke.github.Requester.lambda$fetch$1(Requester.java:76)
	at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:449)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:399)
	... 39 more
Caused by: java.io.FileNotFoundException: https://api.github.com/organizations/8778468/team/7043227/discussions
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1993)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1589)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient.getResponseInfo(GitHubHttpUrlConnectionClient.java:63)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:387)
	... 39 more
[WARN]    [discussions] org.kohsuke.github.GHFileNotFoundException: https://api.github.com/organizations/8778468/team/7043227/discussions {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/teams#create-a-discussion"}
```

**Comment by aalmiray** [comment]
Since the first release, announcer failures do not cause the release to fail as that would likely rollback published packages. Instead, announcers failures are turned into warnings.

**Comment by charphi** [comment]
By the way, GitHub Discussion ignores the `jreleaser.dry.run` property.

**Comment by charphi** [comment]
> Since the first release, announcer failures do not cause the release to fail as that would likely rollback published packages. Instead, announcers failures are turned into warnings.

I agree, that's the right behavior.
The problem is that the warning is not clear:  
`[WARN]    [discussions] org.kohsuke.github.GHFileNotFoundException: https://api.github.com/organizations/8778468/team/7043227/discussions {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/teams#create-a-discussion"}`

**Comment by aalmiray** [comment]
The last element you posted from the log shows that the `discussions` announcer outputs a warning.

**Comment by aalmiray** [comment]
Indeed. There's no easy way at the moment to map lack of token permissions to a proper error message as the failure could occur because of other missconfigured values.

**Comment by aalmiray** [comment]
Released in v1.4.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.4.0
